### PR TITLE
variable length access tokens

### PR DESCRIPTION
**Pull request by lgarvey**
Users on trusted devices, e.g. devices accessing the sites from secure networks with known IP ranges are given longer sessions.

### CODE REVIEW

File: config/settings.py [hunk 1]
```diff
@@ -313,3 +317,5 @@
 # This is used to determine a users primary email and should be set to a comma separated list
 # e.g: 'mobile.ukti.gov.uk, digital.trade.gov.uk'
 DEFAULT_EMAIL_ORDER = env('DEFAULT_EMAIL_ORDER', default="")
+
+TRUSTED_IPS = env('TRUSTED_IPS', default='127.0.0.1')
```
Comment (michalc) [inline config/settings.py]: Minor thing, but I suspect better to have no default, and raise an exception if TRUSTED_IPS is not set There is a bit of a gotcha: if it ends up determining IP by REMOTE_ADDR, which is often 127.0.0.1 in PaaS(?), then not setting it (e.g. accidentally) would mean every single request is trusted.

File: sso/oauth2/ip_check.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        return False
+
+    return ip_in_range(client_ip, trusted_ips=trusted_ips)
+
+
+def ip_in_range(client_ip, trusted_ips):
+    for ip in trusted_ips:
+        if is_cidr(ip) and ipaddress.ip_address(client_ip) in ipaddress.ip_network(ip):
+            return True
+        elif client_ip == ip:
```
Comment (michalc) [inline sso/oauth2/ip_check.py]: A (minor) simplification/reduction of logic would be to only allow CIDRs, so a fixed IP would end in /32

File: sso/oauth2/ip_check.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import re
+
+from ipware.ip import get_real_ip
+from django.conf import settings
+
+
+log = logging.getLogger(__name__)
+
+
+def is_trusted_ip(request, trusted_ips=None):
```
Comment (michalc) [inline sso/oauth2/ip_check.py]: I think the second parameter is only ever passed in tests, and not in production code. Better for this function to not allow a second argument? (... or, _always_ enforce two arguments...)

File: sso/oauth2/ip_check.py (hunk 2 above)
Comment (michalc) [reply sso/oauth2/ip_check.py]: ... actually, I would then also see how `ipaddress.ip_network(` behaves when passed a bad CIDRs. You might not even need the `is_cidr` check at all.

File: config/settings.py (hunk 1 above)
Comment (lgarvey) [reply config/settings.py]: This is a fairly old PR - unsure if ti will be revived or not. If it is, it'll need a fair amount of reworking.

**Pull request closed without merging**